frontend/components/SetupPage/Create/SetupEoaFunding.tsx

```diff
@@ -135,135 +126,128 @@ export const SetupEoaFundingForChain = ({
+    [EvmChainId.Base]: {
+      provider: PROVIDERS[EvmChainId.Base].provider,
+      chainConfig: CHAIN_CONFIG[EvmChainId.Base],
+      requiredEth: MIN_ETH_BALANCE_THRESHOLDS[EvmChainId.Base].safeCreation,
+    },
+  },
+  // [AgentType.Optimus]: {
+  //   [EvmChainId.Optimism]: {
+  //     provider: PROVIDERS[EvmChainId.Optimism].provider,
+  //     chainConfig: CHAIN_CONFIG[EvmChainId.Optimism],
+  //     requiredEth: MIN_ETH_BALANCE_THRESHOLDS[EvmChainId.Optimism].safeCreation,
   //   },
-  //   [MiddlewareChain.ETHEREUM]: {
-  //     provider: ETHEREUM_PROVIDER,
-  //     chainConfig: CHAIN_CONFIG.ETHEREUM,
-  //     requiredEth:
-  //       MIN_ETH_BALANCE_THRESHOLDS[MiddlewareChain.ETHEREUM].safeCreation,
+  //   [EvmChainId.Ethereum]: {
+  //     provider: PROVIDERS[EvmChainId.Ethereum].provider,
+  //     chainConfig: CHAIN_CONFIG[EvmChainId.Ethereum],
+  //     requiredEth: MIN_ETH_BALANCE_THRESHOLDS[EvmChainId.Ethereum].safeCreation,
   //   },
-  //   [MiddlewareChain.BASE]: {
-  //     provider: BASE_PROVIDER,
-  //     chainConfig: CHAIN_CONFIG.BASE,
-  //     requiredEth: MIN_ETH_BALANCE_THRESHOLDS[MiddlewareChain.BASE].safeCreation,
+  //   [EvmChainId.Base]: {
+  //     provider: PROVIDERS[EvmChainId.Base].provider,
+  //     chainConfig: CHAIN_CONFIG[EvmChainId.Base],
+  //     requiredEth: MIN_ETH_BALANCE_THRESHOLDS[EvmChainId.Base].safeCreation,
   //   },
-  [MiddlewareChain.GNOSIS]: {
-    provider: PROVIDERS[EvmChainId.Gnosis].provider,
-    chainConfig: CHAIN_CONFIG[EvmChainId.Gnosis],
-    requiredEth: MIN_ETH_BALANCE_THRESHOLDS[EvmChainId.Gnosis].safeCreation,
-  },
+  // }
 };
 
 export const SetupEoaFunding = () => {
   const { goto } = useSetup();
+  const { selectedAgentType, selectedAgentConfig } = useServices();
   const { masterEoa } = useMasterWalletContext();
   const { masterWalletBalances } = useMasterBalances();
   const masterEoaAddress = masterEoa?.address;
 
-  const [currentChain, setCurrentChain] = useState<MiddlewareChain>(
-    MiddlewareChain.GNOSIS,
+  const [currentChain, setCurrentChain] = useState<EvmChainId>(
+    selectedAgentConfig.evmHomeChainId,
   );
 
-  const currentFundingMapObject = eoaFundingMap[MiddlewareChain.GNOSIS];
+  const currentFundingMapObject =
+    EOA_FUNDING_MAP[selectedAgentType][currentChain];
 
-  const getIsCurrentChainFunded = useCallback(
-    async (
-      currentFundingMapObject: (typeof eoaFundingMap)[keyof typeof eoaFundingMap],
-      masterEoaAddress: Address,
-    ) => {
-      const { provider, requiredEth } = currentFundingMapObject;
-
-      return provider
-        .getBalance(masterEoaAddress)
-        .then(
-          (balance: BigNumber) =>
-            parseFloat(ethers.utils.formatEther(balance)) >= requiredEth,
-        );
-    },
-    [],
+  const eoaBalance = masterWalletBalances?.find(
+    (balance) =>
+      balance.walletAddress === masterEoaAddress &&
+      balance.evmChainId === currentChain,
   );
 
-  useInterval(async () => {
```

Comment:
I decided to replace useInterval with just useEffect, because we fetch balances in background anyway, but lmkyt

Reply:
makes sense; this part was rushed during optimus poc, so likely badly implemented previously